app/src/main/java/com/snatik/matches/common/Memory.java

```diff
@@ -3,26 +3,81 @@
 import android.content.Context;
 import android.content.SharedPreferences;
 import android.content.SharedPreferences.Editor;
+import android.preference.PreferenceManager;
+import android.util.Log;
+
+import java.util.Map;
+
+import static android.R.attr.id;
+import static android.R.id.edit;
+import static android.content.Context.MODE_PRIVATE;
 
 
 public class Memory {
 
 	private static final String SHARED_PREFERENCES_NAME = "com.snatik.matches";
 	private static String highStartKey = "theme_%d_difficulty_%d";
+	private static String bestTimeKey = "themeTime_%d_difficultyTime_%d";
 
 	public static void save(int theme, int difficulty, int stars) {
+		Log.i("In save()","");
 		int highStars = getHighStars(theme, difficulty);
+
 		if (stars > highStars) {
-			SharedPreferences sharedPreferences = Shared.context.getSharedPreferences(SHARED_PREFERENCES_NAME, Context.MODE_PRIVATE);
-			Editor edit = sharedPreferences.edit();
+			Log.i("New best stars","");
+			SharedPreferences sharedPreferences = Shared.context.getSharedPreferences(SHARED_PREFERENCES_NAME, MODE_PRIVATE);
+			SharedPreferences.Editor edit = sharedPreferences.edit();
 			String key = String.format(highStartKey, theme, difficulty);
 			edit.putInt(key, stars).commit();
 		}
 	}
 
+	public static void saveTime(int theme,int difficulty,int passedSecs) {
+
+		int bestTime = getBestTime(theme,difficulty);
+
+		if(passedSecs<bestTime && bestTime != -1) {
+			SharedPreferences sharedPreferences = Shared.context.getSharedPreferences(SHARED_PREFERENCES_NAME,MODE_PRIVATE);
+			SharedPreferences.Editor editor = sharedPreferences.edit();
+			String timeKey = String.format(bestTimeKey, theme, difficulty);
+			editor.putInt(timeKey,passedSecs);
+			editor.commit();
+		}
+		if( bestTime == -1) {
```

Comment:
Remove both *if statements* and change it to one: 
```
if (passedSecs < bestTime || bestTime == -1) {
   ...
}
```
This will cover both cases.
1. if it's the first time then `bestTime == -1`
2. if it's the second time then we check `passedSecs < bestTime`